### PR TITLE
Document flash retrying

### PR DESCRIPTION
**Pull request by juhhov**
## Status
**READY**

## Description
Add documentation for flash retry feature.
Add also logic fix, which removes unnecessary sleep after last retry round.

### CODE REVIEW

File: mbed_flasher/common.py [hunk 1]
```diff
@@ -131,11 +131,14 @@ def retry(logger, func, func_args, retries=DEFAULT_RETRY_AMOUNT, conditions=None
...
                 raise error
 
-        index_from_one = index + 1
-        retry_interval = index_from_one ** 2
-        logger.info("Starting retry round {} after {} sleep"
-                    .format(index_from_one, retry_interval))
-        time.sleep(retry_interval)
+            index_from_one = index + 1
+            if index_from_one == retries:
+                raise error
```
Comment (jupe) [inline mbed_flasher/common.py]: this makes sense but looks that for loop now returns or raises every time - Is there anymore any path which goes to lines after for loop: https://github.com/ARMmbed/mbed-flasher/blob/e18de65e2ca7d7c7c554f524c9522fedde6f5e5c/mbed_flasher/common.py#L143-L147

Comment (juhhov) [reply mbed_flasher/common.py]: Those lines can be removed now.

File: mbed_flasher/common.py [hunk 2]
```diff
@@ -119,29 +119,24 @@ def retry(logger, func, func_args, retries=DEFAULT_RETRY_AMOUNT, conditions=None
     :param conditions: conditions on when to retry
     :return: latest return code
```
Comment (YifeiZuo01) [inline mbed_flasher/common.py]: docstring update? no `return_value` anymore

Comment (juhhov) [reply mbed_flasher/common.py]: On success case it will return normally, line 127.

Comment (YifeiZuo01) [reply mbed_flasher/common.py]: okay

**Pull request merged**